TDL-06-EC2-Cloudwatch-S3/terraform-code-ec2-cw-s3/iam.tf

```diff
@@ -0,0 +1,61 @@
+resource "aws_iam_role" "ec2-role" {
+  name               = "${var.project_name}-ec2-s3-cwlogs-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": "InstanceS3CWRole"
+    }
+  ]
+}
+EOF
+
+  tags = {
+    Name = "${var.project_name}-ec2-s3-cwlogs-role"
+  }
+}
+
+resource "aws_iam_instance_profile" "role-instanceprofile" {
+  name = "${var.project_name}-ec2-s3-cwlogs-role-instanceprofile"
+  role = aws_iam_role.ec2-role.name
+}
+
+resource "aws_iam_role_policy" "role-policy" {
+  name = "${var.project_name}-ec2-s3-cwlogs-role-policy"
+  role = aws_iam_role.ec2-role.id
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:DescribeLogStreams"
+            ],
+            "Resource": [
+                "arn:aws:logs:*:*:*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject"
+            ],
+            "Resource": [
+                "arn:aws:s3:::tdl6-bucket/*"
```

Comment:
As u have the S3bucket name defined in ur variable,  it is better to refer the S3 bucket name to make ur code reusable.